app/questionnaire_state/user_journey_manager.py

```diff
@@ -63,22 +77,33 @@ def go_to_state(self, item_id):
             self._create_new_state(item_id)
 
     def _create_new_state(self, item_id):
-        item = self._schema.get_item_by_id(item_id)
+
         logger.debug("Creating new state for %s", item_id)
-        if isinstance(item, SchemaBlock):
-            state = StateBlock.construct_state(item)
+        if item_id in self._valid_locations:
```

Comment:
I wonder if there is any way we can remove this case style statement and use polymorphism? 
http://refactoring.com/catalog/replaceConditionalWithPolymorphism.html